### PR TITLE
terranix: use wrapProgram on terranix-doc-json

### PR DESCRIPTION
**Pull request by thiagokokada**
###### Motivation for this change

`wrapProgram` does work in scripts.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by thiagokokada** [comment]
CC @mrVanDalo.

**Comment by thiagokokada** [comment]
Resulting wrapper:

```
$ cat ./result/bin/terranix-doc-json
#! /nix/store/phqa311klldrcbwid1i22dwnpfc9dnma-bash-5.1-p8/bin/bash -e
export PATH='/nix/store/22r0xq2brgk0xha0djg0s3sznhrpa5hf-jq-1.6-bin/bin:/nix/store/idx1xbwdhm37gbrk2nnqqi1x5q98hlxn-nix-2.3.16/bin'${PATH:+':'}$PATH
exec -a "$0" "/nix/store/ln5cbngny2fr4b24b947bd6wvhixp6rd-terranix-2.5.0/bin/.terranix-doc-json-wrapped"  "$@"
```

File: pkgs/applications/networking/cluster/terranix/default.nix [hunk 1]
```diff
@@ -11,20 +11,14 @@ stdenv.mkDerivation rec {
...
-    # manual wrapper because makeWrapper expectes executables
-    wrapper=$out/bin/terranix-doc-json
-    cat <<EOF>$wrapper
-    #!/usr/bin/env bash
-    export PATH=$PATH:${jq}/bin:${nix}/bin
-    $out/bin/.wrapper_terranix-doc-json "\$@"
-    EOF
-    chmod +x $wrapper
+    wrapProgram $out/bin/terranix-doc-json \
+      --prefix PATH : ${lib.makeBinPath [jq nix]}
```
Comment (SuperSandro2000) [inline pkgs/applications/networking/cluster/terranix/default.nix:21]:
```suggestion
      --prefix PATH : ${lib.makeBinPath [ jq nix ]}
```

Comment (thiagokokada) [reply pkgs/applications/networking/cluster/terranix/default.nix]: Oops, my fault for not using Emacs to do this PR :sweat_smile: .

Comment (thiagokokada) [reply pkgs/applications/networking/cluster/terranix/default.nix]: Done.

File: pkgs/applications/networking/cluster/terranix/default.nix [hunk 2]
```diff
@@ -1,4 +1,4 @@
-{ stdenv, lib, fetchFromGitHub, jq, nix, ... }:
+{ stdenv, lib, fetchFromGitHub, jq, nix, makeWrapper, ... }:
```
Comment (SuperSandro2000) [inline pkgs/applications/networking/cluster/terranix/default.nix:1]:
```suggestion
{ stdenv, lib, fetchFromGitHub, jq, nix, makeWrapper }:
```

Comment (thiagokokada) [reply pkgs/applications/networking/cluster/terranix/default.nix]: Done.

**Comment by thiagokokada** [comment]
I tested the wrapped program and it seems to work fine, also the wrapper looks correctly.

Going to merge it.

**Comment by mrVanDalo** [comment]
I tried the same, I got an error, because terranix-doc-json is not a binary.

**Comment by mrVanDalo** [comment]
@thiagokokada yes it seems to work. I don't know what I did wrong here. Thanks for fixing that.

**Pull request merged**